swc/private/swc.bzl

```diff
@@ -133,26 +135,22 @@ def _impl(ctx):
             map_outs = ctx.outputs.map_outs
         else:
             map_outs = _declare_outputs(ctx, _calculate_map_outs(srcs, ctx.attr.source_maps))
-        outputs = []
-        outputs.extend(js_outs)
-        outputs.extend(map_outs)
+        output_sources = js_outs + map_outs
+
         for i, src in enumerate(ctx.files.srcs):
             src_args = ctx.actions.args()
 
             js_out = js_outs[i]
-            inputs = [copy_file_to_bin_action(ctx, src)] + ctx.toolchains["@aspect_rules_swc//swc:toolchain_type"].swcinfo.tool_files
+            inputs = [copy_file_to_bin_action(ctx, src)] + swcinfo.tool_files,
             outputs = [js_out]
-            if ctx.attr.source_maps in ["true", "both"]:
+            if map_outs:
```

Comment:
this is a more robust check because the function that creates the map_outs checks the other 2 enum values instead